src/Components/Components/src/RenderTree/RenderTreeDiffBuilder.cs

```diff
@@ -951,16 +951,8 @@ private static void InitializeNewSubtree(ref DiffContext diffContext, int frameI
     private static void InitializeNewComponentFrame(ref DiffContext diffContext, int frameIndex)
     {
         var frames = diffContext.NewTree;
-        ref var frame = ref frames[frameIndex];
-
-        if (frame.ComponentStateField != null)
-        {
-            throw new InvalidOperationException($"Child component already exists during {nameof(InitializeNewComponentFrame)}");
-        }
```

Comment:
This check was redundant since `InstantiateChildComponentOnFrame` already does that.